update README and sync submodule (#20)
Simple-Robotics/proxsuite: README.md

Alternative installation procedures are presented in the [Installation Procedure](#installation-procedure) section.

> ## Proxsuite main features

Reviewer: ```suggestion
## ProxSuite main features
```

**Proxsuite** is fast:

   - C++ template library,
   - cache friendly.

> **Proxsuite** is versatile, offering through simple API advanced algorithms specialized for covering efficiently generic problem structures:

Reviewer: ```suggestion
**Proxsuite** is versatile, offering through a unified API advanced algorithms specialized for efficiently exploiting problem structures:
```

   - dense, sparse and matrix free matrix factorization backends,

>    - specialized warm starts (e.g., equality constrained initial guess, warm start or cold start options using previous results).

Reviewer: ```suggestion
   - advanced warm-starting options (e.g., equality-constrained initial guess, warm-start or cold-start options from previous results).
```

**Proxsuite** is flexible:

   - header only,

>    - C++ 17/20 compliant.

Reviewer: ```suggestion
   - C++ 17/20 compliant,
   - Python and Julia bindings for easy code prototyping without sacrificing performances.
```

**Proxsuite** is extensible.

> **Proxsuite** is reliable and extensively tested.

Reviewer: ```suggestion
**Proxsuite** is reliable and extensively tested, showing the best performances on the hardest problems of the literature.
```
